docs/scripting-element-object.rst

```diff
@@ -0,0 +1,393 @@
+.. _splash-element:
+
+Element Object
+==============
+
+Element objects are instanced by :ref:`splash-select`;
+
+.. _splash-element-attributes:
+
+Attributes
+~~~~~~~~~~
+
+The following fields are read-only.
+
+.. _splash-element-id:
+
+element.id
+----------
+
+Id of the inner representation of the element.
+
+Methods
+~~~~~~~
+
+To modify or retrieve some information about the element you can use the following methods.
+Note that all of the following methods return flag indicating whether the operation was
+successful or not. The methods returns ``ok, reason_or_value`` pair. If ``ok`` is nil
+then error happened during the operation; ``reason`` provides an information about error type;
+otherwise ``ok`` is ``true`` and the returned value is stored in the second variable.
+
+
+.. _splash-element-exists:
+
+element:exists
+--------------
+
+Check whether the element exists in DOM. If the element doesn't exist almost all other
+methods will fail.
+
+**Signature:** ``exists = element:exists()``
+
+**Returns:** ``exists`` indicated whether the element exists.
+
+**Async:** no.
+
+
+.. _splash-node-property:
+
+element:node_property
+---------------------
+
+Return the value of the specified property of the element
+
+**Signature:** ``ok, value = element:node_property(property_name)``
+
+**Parameters:**
+
+* property_name - name of the node property
+
+**Returns:** ``ok, value`` pair. If ``ok`` is nil then error happened during the function
+call; ``value`` provides an information about error type; otherwise ``value`` is a value of
+the specified property
+
+**Async:** no.
+
+
+.. _splash-node-method:
+
+element:node_method
+-------------------
+
+Return the callable Lua function which call the specified node method.
+
+**Signature:** ``ok, method = element:node_method(method_name)``
+
+**Parameters:**
+
+* method_name - name of the node method
+
+**Returns:** ``ok, method`` pair. If ``ok`` is nil then error happened during the function
+call; ``method`` provides an information about error type; otherwise ``value`` is a function
+that can be called from Lua to execute node method in page context.
+
+**Async:** no.
+
+See more in :ref:`splash-jsfunc`.
+
+.. note::
+
+    If the specified method returns another HTML element it will be represented as a table with
+    ``type`` property which equals to ``'node'``
+
+
+.. _splash-element-mouse-click:
+
+element:mouse_click
+-------------------
+
+Trigger mouse click event on the top-left corner of the element.
+
+**Signature:** ``ok, reason = element:mouse_click()``
```

Comment:
Ok
What if that coordinate exceed the element width or height? Just ignore it and click on the other element or throw an error?